cub/util_namespace.cuh

```diff
@@ -108,6 +108,60 @@
 #define CUB_NS_QUALIFIER ::cub
 #endif
 
+#define CUB_COUNT_N(_1, _2, _3, _4, _5, _6, _7, _8, _9, _10, _11, _12, _13, _14, _15, N, ...) N
+#define CUB_COUNT(...) \
+  CUB_IDENTITY(CUB_COUNT_N(__VA_ARGS__, 15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1))
```

Comment:
This will be needed to support 20. As it currently stands compiling with `-arch=all` will generate 14 entries for x86-64 machines. With a couple of new generations of hardware we will easily overflow the 15 count limit.

Reply:
Hm, I wonder if there is some clever way to collapse all the archs down into a single number or some other condensed representation.

Reply:
@jrhemstad having namespace name readable has advantage in my understanding. If you have a binary, you can see why you have multiple symbols. Can you share a value in the single number representation?